pandas/_libs/tslibs/offsets.pyx

```diff
@@ -126,6 +130,26 @@ def apply_index_wraps(func):
     return wrapper
 
 
+cdef _wrap_timedelta_result(result):
+    """
+    Tick operations dispatch to their Timedelta counterparts.  Wrap the result
+    of these operations in a Tick if possible.
+
+    Parameters
+    ----------
+    result : object
+
+    Returns
+    -------
+    object
+    """
+    if PyDelta_Check(result):
+        # convert Timedelta back to a Tick
+        from pandas.tseries.offsets import _delta_to_tick
```

Comment:
this would be easy in cython, except for the fact that the offsets are in python :<

Reply:
Yah that and to_offset; im pretty sure we’ll finish moving the offsets to cython eventually (may need to split this module)